### PR TITLE
Rollback support, Also other stuff.. uuh woops

### PR DESCRIPTION
**Pull request by JSaurusRex**
Added support for rollback which is disabled per player by default with /rollback (since it also affects others)
also added antizoom (previous behaviour was always on, now its an option thats off by default because who uses zoom hacks nowadays?) and it hurts high ping players.

also fixed join and switching team messages

## Checklist

- [X] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/server/entities/character.h [hunk 1]
```diff
@@ -26,6 +26,8 @@ class CCharacter : public CEntity
 	MACRO_ALLOC_POOL_ID()
 
 public:
+	#define POSITION_HISTORY 50
```
Comment (hardliner66) [inline src/game/server/entities/character.h]: No. Just, no. Move that out of the class or make it a static member.

File: src/game/server/entities/flag.cpp [hunk 2]
```diff
@@ -22,6 +22,8 @@ void CFlag::Reset()
 	m_Pos = m_StandPos;
 	m_Vel = vec2(0,0);
 	m_GrabTick = 0;
+	for(int i = 0; i < 50; i++)
```
Comment (hardliner66) [inline src/game/server/entities/flag.cpp]: Use define.

File: src/game/server/entities/flag.h [hunk 3]
```diff
@@ -14,6 +14,8 @@ class CFlag : public CEntity
 	vec2 m_Vel;
 	vec2 m_StandPos;
 
+	vec2 m_Positions[50];
```
Comment (hardliner66) [inline src/game/server/entities/flag.h]: Use define.

File: src/game/server/entities/character.h (hunk 1 above)
Comment (JSaurusRex) [reply src/game/server/entities/character.h]: ;)

**Comment by hardliner66** [comment]
LGTM

**Pull request merged**